src/journal/bsod.c

```diff
@@ -0,0 +1,143 @@
+/* SPDX-License-Identifier: LPGL-2.1-or-later */
+
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "build.h"
+#include "fd-util.h"
+#include "log.h"
+#include "main-func.h"
+#include "sd-id128.h"
+#include "sd-journal.h"
+#include "terminal-util.h"
+
+static void help(void) {
+        printf("%s\n\n"
+               "filters the journal to fetch the first message from the current boot with an emergency log level.\n\n"
+               "   -h --help            Show this help\n"
+               "      --version         Show package version\n",
+               program_invocation_short_name);
+}
+
+static char* first_emerg_boot_message(void) {
```

Comment:
please return the error code here as we usually do, i.e. change the prototype to:

```c
static int acquire_first_emergency_log_message(char **ret) {
```

or so